python/ray/tests/test_channel.py

```diff
@@ -148,11 +151,16 @@ def pass_channel(self, channel):
         def close(self):
             self._channel.close()
 
-        def write(self):
-            self._channel.write(b"x")
+        def write(self, arr):
+            self._channel.write(arr)
 
         def read(self):
-            self._channel.read()
+            self.arr = self._channel.read()
+            # Keep self.arr in scope. This is to check that the next time we
```

Comment:
Would you mind explaining more about this? I can't fully understand the comments.

Reply:
Yes, this is checking that even though the buffer lock is acquired (because self.arr is still in scope), we should still allow the next `read()` to finish (because it should throw an error anyway).

Reply:
Can you update the comment to clarify that ReadRelease() (in C++) is not called on the object while it is still in scope?

Reply:
When will ReadRelease be called? It seems to be called in the MutableObjectBuffer's destructor. What is the relationship between `self.arr` and `MutableObjectBuffer`? Is the lifecycle of `MutableObjectBuffer` managed by reference counting?

https://github.com/ray-project/ray/blob/71d9084a5cdddc183c9a60958ea6440b16f25e63/src/ray/core_worker/experimental_mutable_object_manager.h#L52-L54

Reply:
Thanks, yes, check the code comment now for more explanation.